components/plugin_marketplace/marketplace_modal.js

```diff
@@ -57,26 +137,22 @@ export default class MarketplaceModal extends React.Component {
     componentDidMount() {
         trackEvent('plugins', 'ui_marketplace_opened');
 
-        this.getMarketplacePlugins();
+        this.fetchPlugins();
     }
 
     componentDidUpdate(prevProps) {
-        if (!isEqual(this.props.pluginStatuses, prevProps.pluginStatuses)) {
-            this.getMarketplacePlugins();
+        // Automatically refresh the component when a plugin is installed or uninstalled.
+        if (this.props.pluginStatuses !== prevProps.pluginStatuses) {
+            this.fetchPlugins();
```

Comment:
Do we need to do this anymore? I saw that the fetch is already called in the action when it completes the install at https://github.com/mattermost/mattermost-webapp/blob/746317dc686d10e59d1e7986784603433ac3a6e9/actions/marketplace.js#L64

Maybe for uninstall but for install now it seems it would be fetching it twice

Reply:
It would indeed fetch twice, but right now this also handles the case of a plugin being installed from a different tab. If we relied solely on this, the update tends to occur too slowly, making the UI seem "wrong" for a while.